core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/PluginTypeChecker.scala

```diff
@@ -109,70 +113,119 @@ class PluginTypeChecker(platform: Platform = PlatformFactory.createInstance(),
 
   private def readSupportedExecs: Map[String, String] = {
     val source = Source.fromResource(SUPPORTED_EXECS_FILE)
-    readSupportedOperators(source)
+    readOperators(source, "execs", true, processOperatorLine)
   }
 
   private def readSupportedExprs: Map[String, String] = {
     val source = Source.fromResource(SUPPORTED_EXPRS_FILE)
     // Some SQL function names have backquotes(`) around their names,
     // so we remove them before saving.
-    readSupportedOperators(source, "exprs").map(
+    readOperators(source, "exprs", true, processOperatorLine).map(
       x => (x._1.toLowerCase.replaceAll("\\`", "").replaceAll(" ",""), x._2))
   }
 
+  def readUnsupportedOpsByDefaultReasons: Map[String, String] = {
+    val execsSource = Source.fromResource(SUPPORTED_EXECS_FILE)
+    val unsupportedExecsBydefault = readOperators(execsSource, "execs", false, processOperatorLine)
+    val exprsSource = Source.fromResource(SUPPORTED_EXPRS_FILE)
+    val unsupportedExprsByDefault = readOperators(exprsSource, "exprs", false, processOperatorLine).
+        map(x => (x._1.toLowerCase.replaceAll("\\`", "").replaceAll(" ",""), x._2))
+    unsupportedExecsBydefault ++ unsupportedExprsByDefault
+  }
+
   private def readSupportedTypesForPlugin: (
       Map[String, Map[String, Seq[String]]], ArrayBuffer[String]) = {
     val source = Source.fromResource(DEFAULT_DS_FILE)
     readSupportedTypesForPlugin(source)
   }
 
-  // operatorType can be exprs, score or execs(default). Reads the columns in file depending
-  // on the operatorType passed to this function.
-  private def readSupportedOperators(source: BufferedSource,
-      operatorType: String = "execs"): Map[String, String] = {
-    val supportedOperators = HashMap.empty[String, String]
+  /**
+   * Reads operators from a source and processes them based on the provided logic.
+   *
+   * @param source The source from which to read the operators.
+   * @param operatorType The type of operators being read ("execs", "exprs" or "score").
+   * @param isSupported Flag to determine if we are reading supported or unsupported operators.
+   * @param processLine A function that takes a line, it's operator type and isSupported flag,
+   *                    and returns a sequence of key-value pairs to add to the result map.
+   * @return A Map containing the processed operators.
+   */
+  def readOperators(source: BufferedSource, operatorType: String, isSupported: Boolean,
+      processLine: (Array[String], String, Boolean) => Seq[(String, String)]
```

Comment:
All use cases of this method call the same function for `processLine` - `processOperatorLine`. In this case, should we remove this as argument?